### PR TITLE
[api] fix enforced package deletion ignoring devel defintion

### PR DESCRIPTION
**Pull request by adrianschroeter**
issue#1256

### CODE REVIEW

File: src/api/test/functional/source_controller_test.rb [hunk 1]
```diff
@@ -1140,6 +1140,20 @@ def test_delete_project_with_local_devel_packages
...
+    delete '/source/home:tom:project/A'
+    assert_response 400
+    assert_select "status", code: "delete_error" do
+      assert_select "summary", "Package is used by following packages as devel package: home:tom:project/B"
+    end
+    delete '/source/home:tom:project/A?force=1'
+    assert_response :success
+    get '/source/home:tom:project/B/_meta'
+    assert_response :success
+    assert_no_xml_tag tag: "devel"
```
Comment (bgeuken) [inline src/api/test/functional/source_controller_test.rb]: Could you create a separate test method for this please?

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -1034,6 +1035,14 @@ def remove_linked_packages
     BackendPackage.where(links_to_id: self.id).delete_all
   end
 
+  def remove_devel_packages
+    Package.where(develpackage: self).each do |devel_package|
+      devel_package.develpackage = nil
```
Comment (hennevogel) [inline src/api/app/models/package.rb]: No permission check and nothing? Hm...

Comment (adrianschroeter) [inline src/api/app/models/package.rb]: On Montag, 19. Oktober 2015, 01:35:26 CEST wrote Henne Vogelsang:
> > @@ -1034,6 +1035,14 @@ def remove_linked_packages
> >      BackendPackage.where(links_to_id: self.id).delete_all
> >    end
> >  
> > +  def remove_devel_packages
> > +    Package.where(develpackage: self).each do |devel_package|
> > +      devel_package.develpackage = nil
> 
> No permission check and nothing? Hm...

yes, it was always that way. The $other maintainer had always the right
to remove this. Just that we had a broken database now after some refactor
submissions.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imend�rffer, Jane Smithard, Graham Norton, HRB 21284 (AG N�rnberg) 
 
Maxfeldstra�e 5                         
90409 N�rnberg 
Germany

Comment (hennevogel) [inline src/api/app/models/package.rb]: Broken Database? The links in the package that points to self should be broken and not silently discarded IMHO

Comment (adrianschroeter) [inline src/api/app/models/package.rb]: On Montag, 19. Oktober 2015, 01:38:19 CEST wrote Henne Vogelsang:
> > @@ -1034,6 +1035,14 @@ def remove_linked_packages
> >      BackendPackage.where(links_to_id: self.id).delete_all
> >    end
> >  
> > +  def remove_devel_packages
> > +    Package.where(develpackage: self).each do |devel_package|
> > +      devel_package.develpackage = nil
> 
> Broken Database? The links in the package that points to self should be broken and not silently discarded IMHO

After local discussion: This is restoring the old behaviour which was okay so far
for Factory (where this is mainly used). 

We could handle this different, for example like inventing a "deleted" package similar
to the "deleted" repository handling. But that would be another patch

File: src/api/test/functional/source_controller_test.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/test/functional/source_controller_test.rb]: hm, why? it would need to create almost the same setup again and just make the test suite slower, no?

Comment (bgeuken) [inline src/api/test/functional/source_controller_test.rb]: I am looking for ways to make tests more granular. This one isn't actually that big. Still it would be nice to get to smaller, more readable tests.

I am still in shock from looking through channel_maintenance_test...
